boards/BMS/Src/App/states/App_BalanceState.c

```diff
@@ -0,0 +1,67 @@
+#include <stdlib.h>
+#include "states/App_InitState.h"
+#include "states/App_BalanceState.h"
+#include "states/App_AllStates.h"
+
+static void BalanceStateRunOnEntry(struct StateMachine *const state_machine)
+{
+    struct BmsWorld *         world  = App_SharedStateMachine_GetWorld(state_machine);
+    struct BmsCanTxInterface *can_tx = App_BmsWorld_GetCanTx(world);
+    struct Airs *             airs   = App_BmsWorld_GetAirs(world);
+
+    App_CanTx_SetPeriodicSignal_STATE(can_tx, CANMSGS_BMS_STATE_MACHINE_STATE_BALANCE_CHOICE);
+    App_Airs_OpenAirPositive(airs);
+}
+
+static void BalanceStateRunOnTick1Hz(struct StateMachine *const state_machine)
+{
+    App_AllStatesRunOnTick1Hz(state_machine);
+}
+
+static void BalanceStateRunOnTick100Hz(struct StateMachine *const state_machine)
+{
+    struct BmsWorld *         world       = App_SharedStateMachine_GetWorld(state_machine);
+    struct Accumulator *      accumulator = App_BmsWorld_GetAccumulator(world);
+    struct BmsCanRxInterface *can_rx      = App_BmsWorld_GetCanRx(world);
+    struct BmsCanTxInterface *can_tx      = App_BmsWorld_GetCanTx(world);
+
+    if (App_AllStatesRunOnTick100Hz(state_machine))
+    {
+        const struct State *next_state           = App_GetBalanceState();
+        bool                is_charging_complete = false;
+        bool                is_balancing_enabled = true;
+
+        if (App_Accumulator_IsPackBalanced(accumulator) && App_Accumulator_HasCellReachedMaxVoltageTarget(accumulator))
```

Comment:
you should call Is_Pack_Fully_Charged from App_Accumulator. its the same logic

Reply:
yup forgot to update, should be updated now